### PR TITLE
Use lodash instead of underscore

### PR DESCRIPTION
**Pull request by Phocea**
As discussed with @fzaninotto, reopening this PR. I have left the original implementation with a global delcared in the vendor.js since this is the solution I tested.

On latest build I am getting the error _.includes is not a function when navigating into one of my custom pages.
Following several threads on Restangular Git (one of them being mgonto/restangular#1225). I found out that Restangular is now compatible with lodash 4.

ng-admin using underscore is causing incompatibilty since the new webpack dependency version has been merged. Making use of lodash fixes the problems

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -39,7 +39,7 @@
     "restangular": "~1.5.2",
     "textangular": "1.4.3",
     "ui-select": "0.18.1",
-    "underscore": "^1.8.3"
+    "lodash": "4.17.2"
```
Comment (fzaninotto) [inline package.json]: we just need `lodash.debounce`, please don't add the entire `lodash` lib as dep.

File: src/javascripts/vendors.js [hunk 2]
```diff
@@ -12,4 +12,4 @@ require('angular-numeraljs');
 require('angular-ui-bootstrap/dist/ui-bootstrap-tpls');
 require('ng-file-upload');
 
-global._ = require('underscore');
+global._ = require('lodash');
```
Comment (fzaninotto) [inline src/javascripts/vendors.js]: no, this won't work anymore now that we don't have lodash as dependency

Comment (Phocea) [inline src/javascripts/vendors.js]: I know, but adding var debounce = require('lodash.debounce'); into ListLayoutController will right ?

Comment (Phocea) [inline src/javascripts/vendors.js]: ..actually just tested, and if I reduce this to lodash.debounce, it is ok for the core code of ng-admin but it then make all Restangular interceptor code to fail as specified in the doc...
So not sure how this should be handled

**Comment by fzaninotto** [comment]
As seen with @Phocea, `lodash` is used by `Restangular` anyway. Adding `lodash.debounce` is then counterproductive, since we can't reuse the full `lodash`. So it's better to depend on `lodash` than `lodash.debounce`.

**Comment by Phocea** [comment]
Rebased and replaced by https://github.com/marmelab/ng-admin/pull/1268

**Pull request closed without merging**